modules/router/src/services/messaging.ts

```diff
@@ -0,0 +1,40 @@
+import { RouterConfigResponse, IBasicMessaging, Result } from "@connext/vector-types";
+import { getRandomBytes32, NatsBasicMessagingService, MessagingConfig } from "@connext/vector-utils";
+import pino, { BaseLogger } from "pino";
+
+import { config } from "../config";
+
+export interface IRouterMessagingService extends IBasicMessaging {
+  // Specialized broadcast methods
+  publishRouterConfig(config: RouterConfigResponse): Promise<void>;
+}
+
+export class NatsRouterMessagingService extends NatsBasicMessagingService implements IRouterMessagingService {
+  private logger: BaseLogger;
+  constructor(private readonly messagingConfig: MessagingConfig) {
+    super(messagingConfig);
+    this.logger = messagingConfig.logger ?? pino();
+  }
+
+  async publishRouterConfig(config: RouterConfigResponse): Promise<void> {
+    const method = "publishRouterConfig";
+    const methodId = getRandomBytes32();
+    this.logger.debug({ method, methodId, config }, "Method started");
+    const subject = `${this.publicIdentifier}.config`;
+    await this.publish(subject, Result.ok(config).toJson());
+    this.logger.debug({ method, methodId }, "Method complete");
+  }
+}
+
+export const configureSubscriptions = async (
```

Comment:
Why is this function necessary?

Reply:
We will likely use it in the future to setup other nats subscriptions and broadcasts as we add more functionality to the service, similar to the `setupListeners` function, so I decided to add it in.